petclinic-frontend/src/features/customers/components/UpdatePetForm.tsx

```diff
@@ -0,0 +1,160 @@
+  const [pet, setPet] = useState<PetResponseModel | null>(null);
+  const [errors, setErrors] = useState<{ [key: string]: string }>({});
+  const [successMessage, setSuccessMessage] = useState<string>('');
+  const [notFound, setNotFound] = useState<boolean>(false);
+
+  // Fetch pet data on component mount
+  useEffect(() => {
+    const fetchPetData = async (): Promise<void> => {
+      if (petId) {
+        try {
+          const response = await getPet(petId);
+          const petData: PetResponseModel = response.data;
+          setPet({
+            ...petData,
+            birthDate: new Date(petData.birthDate),
+          });
+        } catch (err) {
+          const error = err as { response?: { status: number } };
+          if (error.response && error.response.status === 404) {
+            setNotFound(true);
+          } else {
+            console.error('Error fetching pet data:', error);
+          }
+        }
+      }
+    };
+
+    fetchPetData().catch(error =>
+      console.error('Error in fetchPetData:', error)
+    );
+  }, [petId]);
+
+  // Handle form field changes
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    if (pet) {
+      const { name, type, value, checked } = e.target;
+      setPet({
+        ...pet,
+        [name]: type === 'checkbox' ? (checked ? 'true' : 'false') : value, // Handle checkbox properly
+      });
+    }
+  };
+
+  // Form validation
+  const validate = (): boolean => {
+    const newErrors: { [key: string]: string } = {};
+    if (!pet?.name) newErrors.name = 'Name is required';
+    if (!pet?.weight) newErrors.weight = 'Weight is required';
+    if (!pet?.petTypeId) newErrors.petTypeId = 'Pet Type is required';
+    setErrors(newErrors);
+    return Object.keys(newErrors).length === 0;
+  };
+  // Handle form submission
+
+  const handleSubmit = async (
+    event: FormEvent<HTMLFormElement>
+  ): Promise<void> => {
+    event.preventDefault();
+    if (!validate() || !pet) return;
+    const petRequestData: PetRequestModel = {
+      name: pet.name,
+      birthDate: pet.birthDate,
+      petTypeId: pet.petTypeId,
+      isActive: pet.isActive,
+      weight: pet.weight,
+    };
+
+    try {
+      const response = await updatePet(petId!, petRequestData);
+      if (response.status === 200) {
+        setSuccessMessage('Pet updated successfully!');
+        navigate(`/customers/${pet.ownerId}`);
+      } else {
+        console.error('Error updating pet');
+      }
+    } catch (error) {
+      console.error('Error updating pet:', error);
+    }
+  };
+
+  // Handle cancel button click
+  const handleCancel = (): void => {
+    if (pet) {
+      navigate(`/customers/${pet.ownerId}`);
+    }
+  };
+
+  if (notFound) {
+    return <p>Pet not found. Please check the pet ID and try again.</p>;
+  }
+
+  if (!pet) {
+    return <p>Loading...</p>;
+  }
+
+  return (
+    <div className="update-pet-form">
+      <h1>Edit Pet: {pet.petId}</h1>
```

Comment:
I would make this the pet name definitely, but the rest looks very good and looks to work well!